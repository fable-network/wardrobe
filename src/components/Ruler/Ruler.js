import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const rulerHorizontal = css`
  height: 1px;
  width: 100%;
`;
const rulerVertical = css`
  width: 2px;
  min-width: 2px;
  align-self: stretch;
`;

const Ruler = styled(({ direction, ...otherProps }) => <div {...otherProps} />)`
  display: block;
  background-color: ${p => p.theme.ruler};
  ${p => (p.direction === 'horizontal' ? rulerHorizontal : rulerVertical)};
`;

Ruler.defaultProps = {
  direction: 'horizontal',
};

Ruler.propTypes = {
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
};

/** @component */
export default Ruler;
