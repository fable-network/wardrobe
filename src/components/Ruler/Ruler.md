Ruler component, horizontal by default:

```jsx
<Ruler /> 
```

But also can be vertical, just remember you have to use it inside a `flex` container:

```jsx
const Stack = require('../../layout/Stack').default;
<Stack direction="horizontal">
  <span>Something</span>
  <Ruler direction="vertical" />
  <span>Something else</span> 
</Stack>
```
