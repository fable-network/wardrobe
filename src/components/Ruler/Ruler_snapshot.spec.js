/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import Ruler from './Ruler';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('Ruler Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <Ruler />
      );
      expect(wrapper1).toMatchSnapshot();

    });
  });
});
