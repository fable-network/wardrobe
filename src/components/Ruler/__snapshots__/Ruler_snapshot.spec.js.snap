// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ruler Component snapshots should match snapshot 1`] = `
.c0 {
  display: block;
  background-color: #e7f1f7;
  height: 1px;
  width: 100%;
}

<Ruler
  direction="horizontal"
>
  <div
    className="c0"
  />
</Ruler>
`;
